airflow/notifiers/hipchat.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import requests
+
+from airflow import configuration
+from airflow.notifiers import AirflowNotifier
+from airflow.exceptions import AirflowException
+from airflow.utils.state import State
+
+
+class HipChatAPI(object):

```

Comment:
This class should probably be airflow.hooks.hipchat.HipChatHook, and support pulling the token from the password field of a connection.